MAUI/DataForm/Working-with-DataForm.md

```diff
@@ -0,0 +1,781 @@
+
+## Customize auto generated fields
+
+You can customize or cancel the generated `DataFormItem` by handling the `GenerateDataFormItem` event. This event occurs when the field is auto-generated for public and non-static property of the data object.
+
+{% tabs %}
+{% highlight XAML %}
+
+    <?xml version="1.0" encoding="utf-8" ?>
+    <ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:dataForm="clr-namespace:Syncfusion.Maui.DataForm;assembly=Syncfusion.Maui.DataForm"
+             x:Class="GettingStarted.MainPage">
+    <ContentPage.Content>
+        <dataForm:SfDataForm x:Name="dataForm" GenerateDataFormItem="OnGenerateDataFormItem"/>
+    </ContentPage.Content>
+    </ContentPage>
+
+{% endhighlight %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+
+    }
+{% endhighlight %}
+{% endtabs %}
+
+`GenerateDataFormItemEventArgs.DataFormItem`  property returns the newly created `DataFormItem`.
+`GenerateDataFormItemEventArgs.DataFormGroupItem`  property returns the newly created `DataFormGroupItem`.
+
+## Cancel DataFormItem generation of the data field
+
+You can cancel the specific `DataFormItem` adding to the data form by handling the `GenerateDataFormItem` event or by defining display attribute to avoid the particular data field being displayed.
+
+#### Using attributes
+
+You can set `AutoGenerateField` to `false` for canceling the `DataFormItem` generation.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(AutoGenerateField = false)]
+    public int ID { get; set; } 
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+In the following code, the `DataFormItem` generation for the `MiddleName` property is canceled by setting the `Cancel` property to true.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+            e.Cancel = true;
+
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing property settings
+
+You can change the property of `DataFormItem` in the `GenerateDataFormItem` event.
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "MiddleName")
+            {
+                e.DataFormItem.IsReadOnly = true;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+#### Changing DataFormItem visibility
```

Comment:
### Chane the autogenerated editor  visibility -  3#

You can change the specific editor visibility by using the `IsVisible` property in the `DataFormItem`.

Reply:
title and content updated